src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1550,6 +1550,49 @@ def aks_agentpool_operation_abort(cmd,   # pylint: disable=unused-argument
     )
 
 
+def aks_agentpool_delete_machines(cmd,   # pylint: disable=unused-argument
+                                  client,
+                                  resource_group_name,
+                                  cluster_name,
+                                  nodepool_name,
+                                  machine_names,
+                                  no_wait=False):
+    agentpool_exists = False
+    instances = client.list(resource_group_name, cluster_name)
+    for agentpool_profile in instances:
+        if agentpool_profile.name.lower() == nodepool_name.lower():
+            agentpool_exists = True
+            break
+
+    if not agentpool_exists:
+        raise CLIError(
```

Comment:
Please use the specific error type `ResourceNotFoundError` instead of `CLIError`
```suggestion
        raise ResourceNotFoundError(
```

Reply:
fixed